backend/uclapi/workspaces/occupeye.py

```diff
@@ -791,6 +803,103 @@ def get_survey_sensors_summary(self, survey_ids):
 
         return sensors_data_dict.values()
 
+    def _cache_historical_time_usage_data(self, survey_id, day_count):
+        """
+        Function to cache in Redis the historical usage data over each 10
+        minute time period.
+        day_count is the number of days of historical data (from including
+        yesterday, but not including today) to cache.
+        E.g. if you cache on Tuesday for 7 days, all the data from
+        last Sunday to Monday (e.g. yesterday) inclusive will be cached.
+        This is to support apps which show historical survey usage data.
+        """
+        end_date = datetime.now() - timedelta(days=1)
+        start_date = end_date - timedelta(days=day_count - 1)
+        headers = {
+            "Authorization": self.get_bearer_token()
+        }
+
+        url = (

```

Comment:
`urlencode(...)` does something similar (given k, v pairs it expands them into k1=v1&...) but I think this can work too :)

Reply:
So weirdly I tried using a dictionary and the the `requests` library's `params` parameter but it encoded the lists like this:
```
Weekdays=Monday&Weekdays=Tuesdays&Weekdays=Wednesday&Weekdays=Thursday&Weekdays=Friday
```
It might be possible to just set the array key to `Weekdays[]` but it feels wrong...